lib/active_merchant/billing/gateways/iveri.rb

```diff
@@ -0,0 +1,231 @@
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((CertificateID=\\\")[^\\]*), '\1[FILTERED]').
+          gsub(%r((&lt;PAN&gt;)[^&]*), '\1[FILTERED]').
+          gsub(%r((&lt;CardSecurityCode&gt;)[^&]*), '\1[FILTERED]')
+      end
+
+      private
+
+      def build_xml_envelope(vxml)
+        builder = Nokogiri::XML::Builder.new(:encoding => 'UTF-8') do |xml|
+          xml[:soap].Envelope 'xmlns:xsi' => 'http://www.w3.org/2001/XMLSchema-instance', 'xmlns:xsd' => 'http://www.w3.org/2001/XMLSchema', 'xmlns:soap' => 'http://schemas.xmlsoap.org/soap/envelope/' do
+            xml[:soap].Body do
+              xml.Execute 'xmlns' => 'http://iveri.com/' do
+                xml.validateRequest 'true'
+                xml.protocol 'V_XML'
+                xml.protocolVersion '2.0'
+                xml.request vxml
+              end
+            end
+          end
+        end
+
+        builder.to_xml
+      end
+
+      def build_vxml_request(action, options)
+        builder = Nokogiri::XML::Builder.new do |xml|
+          xml.V_XML('Version' => '2.0', 'CertificateID' => @options[:cert_id], 'Direction' => 'Request') do
+            xml.Transaction('ApplicationID' => @options[:app_id], 'Command' => action, 'Mode' => mode) do
+              yield(xml)
+            end
+          end
+        end
+
+        builder.doc.root.to_xml
+      end
+
+      def add_params_in_order(post, action, money, pm, options)
+        post.Amount(amount(money))                    if money
+        post.Currency(options[:currency])             if [:purchase, :capture, :refund].include? action
+        post.ExpiryDate(expiration(pm))               if pm.respond_to?(:verification_value)
+        add_reference(post, action, pm, options)
+        post.TransactionIndex(split_auth(pm)[1])      if [:capture, :refund].include? action
+        post.OriginalRequestID(split_auth(pm)[0])     if [:void].include? action
+        post.CardSecurityCode(pm.verification_value)  if pm.respond_to?(:verification_value)
+        post.PAN(pm.number)                           if pm.respond_to?(:number)
+      end
+
+      def add_reference(post, action, payment_method, options)
+        if payment_method.is_a?(String)
+          post.MerchantReference(split_auth(payment_method)[2])
+        else
+          post.MerchantReference(options[:order_id] || generate_unique_id)
+        end
+      end
+
+      def expiration(payment_method)
+        "#{format(payment_method.month, :two_digits)}#{payment_method.year}"
+      end
+
+      def commit(post)
+        parsed = parse(ssl_post(live_url, build_xml_envelope(post), headers(post)))
+        succeeded = success_from(parsed)
+
+        Response.new(
+          succeeded,
+          message_from(parsed, succeeded),
+          parsed,
+          authorization: authorization_from(parsed),
+          avs_result: AVSResult.new(code: parsed["some_avs_response_key"]),
```

Comment:
iVeri doesn't support AVS or seem to return CVV "results"(?), so these were untouched defaults. I'm not sure what's best to replace these with in that case... (some other gateways still have these defaults)